cmd/ts-info/app/hevc.go

```diff
@@ -46,69 +47,54 @@ func (a *hevcPS) setPPS(nalu []byte) error {
 	return nil
 }
 
-func parseHEVCPES(jp *jsonPrinter, d *astits.DemuxerData, ps *hevcPS, verbose bool) (*hevcPS, error) {
+func parseHEVCPES(jp *jsonPrinter, d *astits.DemuxerData, ps *hevcPS, o Options) (*hevcPS, error) {
 	pid := d.PID
 	pes := d.PES
+	fp := d.FirstPacket
 	if pes.Header.OptionalHeader.PTS == nil {
 		return nil, fmt.Errorf("no PTS in PES")
 	}
-
 	nfd := naluFrameData{
 		PID: pid,
 	}
-	if d.FirstPacket != nil {
-		af := d.FirstPacket.AdaptationField
-		if af != nil {
-			nfd.RAI = af.RandomAccessIndicator
+	if ps == nil {
+		// return empty PS to count picture numbers correctly
+		// even if we are not printing NALUs
+		ps = &hevcPS{}
+	}
+	pts := *pes.Header.OptionalHeader.PTS
+	nfd.PTS = pts.Base
+	ps.statistics.Type = "HEVC"
+	ps.statistics.Pid = pid
+	if fp != nil && fp.AdaptationField != nil {
+		nfd.RAI = fp.AdaptationField.RandomAccessIndicator
+		if nfd.RAI {
+			ps.statistics.RAIPTS = append(ps.statistics.IDRPTS, pts.Base)
 		}
 	}
-	nfd.PTS = pes.Header.OptionalHeader.PTS.Base
+
 	dts := pes.Header.OptionalHeader.DTS
 	if dts != nil {
-		nfd.DTS = &dts.Base
+		nfd.DTS = dts.Base
+		ps.statistics.TimeStamps = append(ps.statistics.TimeStamps, dts.Base)
+	} else {
```

Comment:
Set nfd.DTS = nfd.PTS if dts is nil, so that we always get a value per line and can more easily check any irregularities.